examples/upload_file.py

```diff
@@ -0,0 +1,60 @@
+import getpass
+import os
+
+from sasctl import Session
+from sasctl.services import files, folders
+
+server_name = "myServer.sas.com"
+user = "sasdemo"
+password = getpass.getpass()
+os_path = "/test/Downloads/pdfs"
+sas_path = "/Public/PDFs"
+
+
+def build_storage_lists(source_path, target_path):
+    """
+    Function to create a list of files and directories
+    from a source folder at OS level
+    and mapped to target folder in SAS Content Server.
+
+    Parameters
+    ----------
+    source_path : str
+        Path to the folder which should be imported.
+    target_path : str
+        Path to the folder which will contain the files and folders in SAS Content Server.
+
+    Returns
+    -------
+    f_list
+        A list of file information which can be used to map local and SAS files.
+    d_list
+        A list of directory information which can be user to map local and SAS folder structure.
+    """
+    f_list = []
+    d_list = []
+
+    for root, _, src_files in os.walk(source_path):
+        for src_file in src_files:
+            if src_file.endswith(".pdf"):
+                if root not in d_list:
```

Comment:
Checking `d_list` for `root` but not actually putting `root` in `d_list` - can result in duplicate entries in `d_list`.  Either make `d_list` a set or check if `dir_info` exists before adding.

Reply:
Changed the different variable names and added a check to validate if folder_info is on folders_list.